src/SprykerEco/Zed/Billpay/Business/Payment/Handler/CancelResponseHandler.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Billpay\Business\Payment\Handler;
+
+use Generated\Shared\Transfer\BillpayCancelResponseTransfer;
+use Generated\Shared\Transfer\OrderTransfer;
+use SprykerEco\Shared\Billpay\BillpaySharedConfig;
+
+class CancelResponseHandler extends AbstractResponseHandler implements CancelResponseHandlerInterface
+{
+    const METHOD = 'CANCEL';
+
+    /**
+     * @param \Generated\Shared\Transfer\BillpayCancelResponseTransfer $responseTransfer
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return void
+     */
+    public function handle(BillpayCancelResponseTransfer $responseTransfer, OrderTransfer $orderTransfer)
+    {
+        $this->logHeader($responseTransfer->getHeader(), self::METHOD);
+
+        if (!$responseTransfer->getHeader()->getIsSuccess()) {
+            return;
+        }
+
+        /** @var \Orm\Zed\Billpay\Persistence\SpyPaymentBillpay $paymentEntity */
+        $paymentEntity = $this
+            ->queryContainer
+            ->queryPaymentByBptid($orderTransfer->getBillpayPayment()->getBptid())
+            ->findOne();
```

Comment:
It can be null, remove inline annotation hack and instead throw meaning ful not found exception maybe?